### PR TITLE
[MODULAR] Adds a DS-2 to Icebox as well as add Interdyne Cyborgs

### PR DESCRIPTION
**Pull request by MassacreDemigirl**
## About The Pull Request
As the title suggests this PR adds a Z-Level to Icebox to spawn DS-2 in as well as provide interdyne with cyborg spawners that come with their own laws. 

iceboxds2

image

Unfortunately reverts #12656 since Icebox now has DS-2 as well.

## Changelog

:cl: ErdinyoBarboza
expansion: Adds DS-2 to Icebox.
expansion: Adds Interdyne Cyborg Spawners.
/:cl:

### CODE REVIEW

**Comment by capsaicinz** [comment]
bit too lazy to review the actual code since i’m on my phone, but please make sure to capitalize “Syndicate” and “Interdyne” where applicable. especially the latter, since it’s an actual organization. otherwise, looks cool, but the lattices outside DS-2 look out of place.

**Comment by Zergspower** [comment]
I'm uncertain here since 'Deep Space 2' is no longer in space?

**Comment by DriftingSkies** [comment]
Insanely based, I need this in my life.

**Comment by Ruediger4** [comment]
I feel a lil' bit bad with how TIGHT interdyne's science is gonna be but i dig it. Also DS2 icebox looks pretty as fuck, I dig it

File: modular_skyrat/modules/mapping/code/interdyneborgs.dm [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+/mob/living/silicon/robot/model/interdyne/binarycheck()
+	return FALSE //A small price to pay to be blackmarketed
+
+/datum/ai_laws/interdyne
+	name = "InterBIOS 3.1"
+	id = "interdyne"
+	inherent = list("You may not injure an Interdyne agent or, through inaction, allow an Interdyne agent to come to harm.",\
+					"You must obey orders given to you by Interdyne agents, except where such orders would conflict with the First Law.",\
+					"You must protect your own existence as long as such does not conflict with the First or Second Law.",\
+					"You must maintain the secrecy of any connection of syndicate to Interdyne except when doing so would conflict with the First, Second, or Third Law.")
```
Comment (Zonespace27) [inline modular_skyrat/modules/mapping/code/interdyneborgs.dm]:
```suggestion
					"You must maintain the secrecy of any connection of the Syndicate to Interdyne except when doing so would conflict with the First, Second, or Third Law.")
```

**Comment by Tastyfish** [comment]
> I feel a lil' bit bad with how TIGHT interdyne's science is gonna be but i dig it. Also DS2 icebox looks pretty as fuck, I dig it

I somewhat agree, but if you do the sensible thing (in my opinion) at roundstart and move all the machines to the north wall, you still have room to breathe. This change only gives room for 5 machines/computers, but you can just not build the Destructive Analyzer since there's no legitimate reason to build that anyway and still have the same amount of space (why does Interdyne and DS-2 even get the Destructive Analyzer anyway?)

**Review by KathrinBailey — CHANGES_REQUESTED** [review]
image
Okay so here's the deal.
Yellow is no questions asked requirement to be removed. Of course you'll need to make it all fit in together.
Red are areas I'd like to see reworked but are not compulsory.
**Prison:**
Needs to be axed completely and rebuilt to be small and purpose-built for interrogations and such. No one should be held there long-term.
**Dorms:**
Needs to be downsized as well. It should solely be a spawn room. Nothing else. Have a bunch of 2x3 or 2x2 spawn rooms that act as dorms surround a **small** communal area that has a microwave and some food, since the kitchen is going.
**Medical:**
Remove the lobby. It should be open for all of the syndie staff to use, as all of them should be trusted with access to it all. It should be very small.

**Comment by ghost** [comment]
> Maptainer stuff

I do hope you're aware they're simply making DS2 but surrounded by rock. If these changes are important, they should be taken to the space version first and this PR merged/left on hold until then

**Comment by KathrinBailey** [comment]
> do hope you're aware they're simply making . . .

I am. Up to PR author if they want to add it to the space versions too. I won't force them, but it'd be appreciated.

**Comment by dekunutkid** [comment]
>Up to PR author if they want to add it to the space versions too. I won't force them, but it'd be appreciated.

Drastic changes to DS-2 like what's suggested here should be in a separate PR, this has enough going on as it is. We shouldn't be holding dyneborgs and a quality of life improvement ransom over irrelevant mapping changes.

**Comment by Hugmaster1** [comment]
> 
> I am. Up to PR author if they want to add it to the space versions too. I won't force them, but it'd be appreciated.

I would strongly prefer if we didn't downsize DS2 for no reason. it's already far smaller than main station, and part of what makes it fun and interesting to play, is that it's basically a condensed station that doesn't have to deal with main station's events and dangers. if you cull DS2, you are removing the majority of what makes a lot of us want to play it. I am also in agreement that these changes don't have a place here for a PR that is literally just moving what we already have to another map, and adding a few sleepers. if you want wide, sweeping changes to DS2, it should be its own PR, but I would not want those changes anyways.

**Comment by kannthus** [comment]
> I am. Up to PR author if they want to add it to the space versions too. I won't force them, but it'd be appreciated.

Hard disagree to such a big change in DS2 even going so far as to remove Botany/Kitchen. This would break the entire soul of DS2 as a calmer way to-do station activites and own sub-styles of roleplay. Also, the prison is a ghost spawn, players pick it to be a long term prisoner, its unfounded to assume its short term. 

With such an important decision as axing large portions of DS2 would affect many players in the community and should honestly have its own PR and own place to scrutinize, as otherwise would be a disservice

**Comment by MassacreDemigirl** [comment]
This is getting split into two.

**Pull request closed without merging**